cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeHandler.java

```diff
@@ -129,32 +119,112 @@ public void metadata(FullHttpRequest request, HttpResponder responder) throws Ex
     Closeables.closeQuietly(chunkResponder);
   }
 
-  private String getTopic(String topicConfig) {
-    int idx = topicConfig.lastIndexOf(':');
-    return idx < 0 ? cConf.get(topicConfig) : cConf.get(topicConfig.substring(0, idx)) + topicConfig.substring(idx + 1);
-  }
-
+  /**
+   * shuts down remote runtime
+   */
   @POST
   @Path("/shutdown")
   public void shutdown(FullHttpRequest request, HttpResponder responder) throws Exception {
     responder.sendString(HttpResponseStatus.OK, "Triggering shutdown down Runtime Http Server.");
     shutdownRunnable.run();
   }
 
-  private void fetchAndWriteMessages(JsonWriter jsonWriter, ByteBuf buffer,
-                                     ChunkResponder chunkResponder, String topic, int limit,
+  /**
+   * Decode consume request from avro binary format
+   */
+  private Map<Utf8, GenericRecord> decodeConsumeRequest(FullHttpRequest request) throws IOException {
```

Comment:
If you don't want to use `Utf8` (which is "heavier" in this particular usage context), you can add a property to the schema by using the `GenericData.setStringType(Schema, StringType)` method, telling avro to use `String` instead of `Utf8`.